tulip_api_auth.html

```diff
@@ -14,16 +14,14 @@
         validate: function (name) {
           const validIpAddr = new RegExp(
             '^(([0-9]|[1-9][0-9]|1[0-9]{2}' +
-              '|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]' +
+              '|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]' +
               '|1[0-9]{2}|2[0-4][0-9]|25[0-5])$',
           );
           const validHostname = new RegExp(
-            '^(([a-zA-Z0-9]|[a-zA-Z0-9]' +
-              '[a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9]' +
-              '[A-Za-z0-9\-]*[A-Za-z0-9])$',
+            '^\\w+([\.\-]\\w+)*$',
```

Comment:
nit: Not going to hurt anything by leaving them in, but the `[\.\-]` here can just be `[.-]` instead. Partly this is because `'\.\-' === '.-'` and you're using single quotes here, but even if the backslashes carried through they'd be unnecessary; a `.` inside a character class (`[.]`) matches a `.` literally, not a wildcard, and a `-` at the end of a character class (`[abc-]`) matches a `-` literally, not a range separator